test/pubsub-in-browser.spec.js

```diff
@@ -0,0 +1,179 @@
+    })
+
+    after((done) => {
+      factory.dismantle(done)
+    })
+
+    describe('everything errors', () => {
+      describe('Callback API', () => {
+        describe('.publish', () => {
+          it('throws an error', (done) => {
+            ipfs1.pubsub.publish(topic, 'hello friend', (err, topics) => {
+              expect(err).to.exist
+              expect(err.message).to.equal(expectedError)
+              done()
+            })
+          })
+        })
+
+        describe('.subscribe', () => {
+          const handler = () => {}
+          it('throws an error', (done) => {
+            ipfs1.pubsub.subscribe(topic, {}, handler, (err, topics) => {
+              expect(err).to.exist
+              expect(err.message).to.equal(expectedError)
+              done()
+            })
+          })
+        })
+
+        describe('.peers', () => {
+          it('throws an error', (done) => {
+            ipfs1.pubsub.peers(topic, (err, topics) => {
+              expect(err).to.exist
+              expect(err.message).to.equal(expectedError)
+              done()
+            })
+          })
+        })
+
+        describe('.ls', () => {
+          it('throws an error', (done) => {
+            ipfs1.pubsub.ls((err, topics) => {
+              expect(err).to.exist
+              expect(err.message).to.equal(expectedError)
+              done()
+            })
+          })
+        })
+      })
+
+      describe('Promise API', () => {
+        describe('.publish', () => {
+          it('throws an error', () => {
+            return ipfs1.pubsub.publish(topic, 'hello friend')
+              .catch((e) => {
+                expect(e).to.exist
+                expect(e.message).to.equal(expectedError)
+              })
+          })
+        })
+
+        describe('.subscribe', () => {
+          const handler = () => {}
+          it('throws an error', (done) => {
+            ipfs1.pubsub.subscribe(topic, {}, handler)
+              .catch((e) => {
+                expect(e).to.exist
+                expect(e.message).to.equal(expectedError)
+                done()
+              })
+          })
+        })
+
+        describe('.peers', () => {
+          it('throws an error', (done) => {
+            ipfs1.pubsub.peers(topic)
+              .catch((e) => {
+                expect(e).to.exist
+                expect(e.message).to.equal(expectedError)
+                done()
+              })
+          })
+        })
+
+        describe('.ls', () => {
+          it('throws an error', () => {
+            return ipfs1.pubsub.ls()
+              .catch((e) => {
+                expect(e).to.exist
+                expect(e.message).to.equal(expectedError)
+              })
+          })
+        })
+      })
+
+      describe('.unsubscribe', () => {
+        it('throws an error', (done) => {
+          try {
+            ipfs1.pubsub.unsubscribe(topic)
+            done('unsubscribe() didn\'t throw an error')
+          } catch (e) {
```

Comment:
Please always use `err`, so that `standard` linter catches if we don't check fot it.